src/main/java/com/mongodb/hibernate/jdbc/MongoConnection.java

```diff
@@ -0,0 +1,221 @@
+        return createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql) {
+        return prepareStatement(mql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+    }
+
+    @Override
+    public Statement createStatement(int resultSetType, int resultSetConcurrency) {
+        throw new NotYetImplementedException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-16");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, int resultSetType, int resultSetConcurrency) {
+        throw new NotYetImplementedException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-13");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, int autoGeneratedKeys) {
+        throw new UnsupportedOperationException("Auto-generated key from MongoDB server side unsupported");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, String[] columnNames) {
+        throw new UnsupportedOperationException("Auto-generated key from MongoDB server side unsupported");
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // SQL99 data types
+
+    @Override
+    public Clob createClob() {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public Blob createBlob() {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public NClob createNClob() {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public SQLXML createSQLXML() {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public Array createArrayOf(String typeName, Object[] elements) {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public Struct createStruct(String typeName, Object[] attributes) {
+        throw new NotYetImplementedException();
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // DataBase meta data
+
+    @Override
+    public DatabaseMetaData getMetaData() {
+        throw new NotYetImplementedException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-37");
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // unsupported operations
+
+    @Override
+    public CallableStatement prepareCall(String mql) {
+        throw new UnsupportedOperationException(
+                "MongoDB's JSON server function has been deprecated in favour of 'aggregate' pipeline");
+    }
+
+    @Override
+    public CallableStatement prepareCall(String mql, int resultSetType, int resultSetConcurrency) {
+        throw new UnsupportedOperationException(
+                "MongoDB's JSON server function has been deprecated in favour of 'aggregate' pipeline");
+    }
```

Comment:
We don't need to support MongoDB's counterpart of SQL's `stored procedure` (as done in Hibernate OGM) for it has been deprecated by aggregate pipeline, as explained at https://www.mongodb.com/resources/products/capabilities/stored-procedures